src/IO/H5/VolumeData.cpp

```diff
@@ -205,6 +207,98 @@ void VolumeData::write_volume_data(
               append_element_extents_and_connectivity(
                   &total_extents, &total_connectivity, &total_points_so_far,
                   dim, element);
+
+              // If element is 2D and the bases are both SphericalHarmonic,
+              // then add extra connections to close the surface.
+              if (dim == 2) {
```

Comment:
Can you move all this connectivity code into a separate free function and test it there? `write_volume_data` doesn't seem like the right place to have the connectivity implementation

Reply:
After thinking about it, I think this code belongs at the bottom of the helper function `append_element_extents_and_connectivity`, so I moved it there